src/Domain/Creators/FrustalCloak.hpp

```diff
@@ -0,0 +1,118 @@
+#include <cstddef>
+#include <vector>
+
+#include "Domain/Domain.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <size_t Dim, typename Frame>
+class DomainCreator;  // IWYU pragma: keep
+/// \endcond
+
+namespace domain {
+namespace creators {
+
+/// \ingroup DomainCreatorsGroup
+/// Create a 3D Domain consisting of frusta. Each frustum is made of two
+/// parallel rectangles framed by four trapezoids.
+template <typename TargetFrame>
+class FrustalCloak : public DomainCreator<3, TargetFrame> {
+ public:
+  struct InitialRefinement {
+    using type = size_t;
+    static constexpr OptionString help = {
+        "Initial refinement level in each dimension."};
+  };
+
+  struct InitialGridPoints {
+    using type = std::array<size_t, 2>;
+    static constexpr OptionString help = {
+        "Initial number of grid points in [r,angular]."};
+  };
+  struct UseEquiangularMap {
+    using type = bool;
+    static constexpr OptionString help = {
+        "Use equiangular instead of equidistant coordinates."};
+    static constexpr type default_value() noexcept { return false; }
+  };
+
+  struct ProjectionFactor {
+    using type = double;
+    static constexpr OptionString help = {"Grid compression factor."};
+    static constexpr type default_value() noexcept { return 1.0; }
+  };
+
+  struct LengthInnerCube {
+    using type = double;
+    static constexpr OptionString help = {"Side length of each inner cube."};
+    static constexpr type default_value() noexcept { return 0.0; }
+  };
+
+  struct LengthOuterCube {
+    using type = double;
+    static constexpr OptionString help = {"Side length of the outer cube."};
+    static constexpr type default_value() noexcept { return 0.0; }
+  };
+
+  using options =
+      tmpl::list<InitialRefinement, InitialGridPoints, UseEquiangularMap,
+                 ProjectionFactor, LengthInnerCube, LengthOuterCube>;
+
+  static constexpr OptionString help{
+      "Creates an excised cube domain using frusta. The parameter\n"
+      "`UseEquiangularMap` can be used to apply a tangent mapping to the xi\n"
+      "and eta logical coordinates, while the parameter `ProjectionFactor` \n"
+      "can be used to apply a projective map to the zeta logical coordinate.\n"
+      "Increasing the `ProjectionFactor` value can give better gridpoint\n"
+      "spacings in the z direction. The user constructs a Domain of ten\n"
+      "frusta by specifying values for `LengthInnerCube` and\n"
+      "`LengthOuterCube`. This will create a cubical Domain of side"
+      "length `LengthOuterCube` with the center excised. The size of the\n"
+      "excised region is determined by the value set for `LengthInnerCube`.\n"
+      "Note that the shape of the excised region is not that of a cube, but\n"
+      "of two abutting cubes. This is because this Domain is primarily for\n"
+      "testing the frustal cloak in the BinaryCompactObject Domain."};
+
```

Comment:
Say that the abutting excised cubes share a face in the x direction (or y or z, whichever it is).